### ISSUE TITLE
Use natural language mode for MySQL Text search

### ISSUE THREAD

**Issue by pantharshit00** [prisma/prisma#16537]
### Discussed in https://github.com/prisma/prisma/discussions/12905

<div type='discussions-op-text'>

<sup>Originally posted by **Nethrenial** April 21, 2022</sup>
I've written a simple text search query with the Prisma client to search Usernames or emails in my users table. While it correctly works with usernames, it errors out saying, `Unexpected '@'`.I believe that's due to @ is a special symbol.Is there anyway around this, because I need to be able to search full email addresses. (Emails are correctly saved in the database). And query return results if I omit the part from @.Ex: in abc@gmail.com, query returns match if I just search abc.But If I use abc@gmail.com, it fails with this error.
```bash
 Error occurred during query execution:
 ConnectorError(
  ConnectorError { 
   user_facing_error: None, 
   kind: QueryError(Server(ServerError { 
                           code: 1064, message: "syntax error, 
                           unexpected '@',expecting $end", state: "42000" }
)) })
```
How do I solve this, I searched the documentation, but I couldn't find anything related to this.
My User model looks like this,
```prisma
model User {
  id              Int     @id @default(autoincrement())
  username        String  @db.VarChar(255)
  email           String  @db.VarChar(255)
  avatar          String? @db.VarChar(255)
  password        String  @default("1") @db.VarChar(255)
  friends         User[]  @relation("friends")
  friendsRelation User[]  @relation("friends")
  hashRT          String? @db.VarChar(255)

  @@fulltext([username])
  @@fulltext([email])
  @@fulltext([username, email])
  @@map("users")
}
```</div>

**Comment by mcbethio** [comment]
I notice that this version is 3.12.0, I have version 4.10.1 and I cannot reproduce this issue.

Is it possible to upgrade and try again on the latest version? Thanks

**Comment by 4815162342** [comment]
@mcbethio That's odd that you're unable to reproduce. I've just reproduced it in my own environment with versions `4.5.0` and `4.10.1`. Both times I get the error the original author posted. 

If it helps, here's my environment info:

```
prisma                  : 4.10.1
@prisma/client          : 4.10.1
Current platform        : darwin
Query Engine (Node-API) : libquery-engine aead147aa326ccb985dcfed5b065b4fdabd44b19 (at ../../../common/temp/node_modules/.pnpm/@prisma+engines@4.10.1/node_modules/@prisma/engines/libquery_engine-darwin.dylib.node)
Migration Engine        : migration-engine-cli aead147aa326ccb985dcfed5b065b4fdabd44b19 (at ../../../common/temp/node_modules/.pnpm/@prisma+engines@4.10.1/node_modules/@prisma/engines/migration-engine-darwin)
Format Wasm             : @prisma/prisma-fmt-wasm 4.10.1-1.80b351cc7c06d352abe81be19b8a89e9c6b7c110
Default Engines Hash    : aead147aa326ccb985dcfed5b065b4fdabd44b19
Studio                  : 0.481.0
Preview Features        : referentialIntegrity, interactiveTransactions, fullTextSearch, fullTextIndex
```

**Comment by mcbethio** [comment]
No worries at all, we will continue investigating...

**Comment by timmeade** [comment]
I just hit this also.   Please advise.

**Comment by yeliex** [comment]
@mcbethio could prisma privide `NATURAL LANGUAGE MODE` for fulltext search, it is useful for non-English content

**Comment by miketromba** [comment]
+1 for the ability to explicitly specify which mode we want to use in MySQL (`NATURAL LANGUAGE MODE` or `BOOLEAN MODE`) - seems like a table-stakes feature to have.

**Comment by miketromba** [comment]
@Ali1Ammar @pantharshit00 @timmeade 

I recommend taking a look at this StackOverflow post: https://stackoverflow.com/a/26537463/6271468

TLDR: the `fulltext` indexes in MySQL don't actually index non-word characters, so the optimal approach is to remove those operator strings via regex by replacing them with spaces or just removing them entirely before running the query.

Note: I have not independently verified this solution yet, so I suggest doing your own experimenting.

**Comment by Druue** [comment]
Hey folks!

I tried this with the following matrix of details:

prisma_version: [3.12.0, 4.10.1, 5.14.0]
mysql_version: [`v5.6`, `v5.7`, `v8`, `MariaDB`]

When I wrapped the search term with two pairs of string quotes I consistently retrieved data with no errors
```
[
  {
    id: 1,
    username: 'username',
    email: 'email@bar',
    avatar: null,
    password: '1',
    hashRT: null
  }
]
```

---

And when I removed the second set of quotes for `search: ` i.e.
```diff
- search: '"foo@bar"'
+ search: "foo@bar"
```

I was able to confirm the same syntax error
```
14 }
  15 
  16 const test = async () => {
→ 17   const res = await prisma.user.findMany(
Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Server(MysqlError { code: 1064, message: "syntax error, unexpected '@', expecting $end", state: "42000" })), transient: false })
 ELIFECYCLE  Command failed with exit code 1.
```
